File: demo04.pl
#!/usr/bin/perl -w
# Reads lines of input until end-of-input
# Print snap! if two consecutive lines are identical
# taken from lecture example <http://www.cse.unsw.edu.au/~cs2041/code/perl/code_examples.html>

$i = 3;
print "1. Enter line:\n";
$last_line = <STDIN>;
print "2. Enter line:\n";
while ($line = <STDIN>) {
    if ($line eq $last_line) {
        print "Snap!\n";
    }
    $last_line = $line;
    print "$i. Enter line:\n";
    $i++;
}

File: demo04.py
#!/usr/local/bin/python3.5 -u
import sys
# Reads lines of input until end-of-input
# Print snap! if two consecutive lines are identical
# taken from lecture example <http://www.cse.unsw.edu.au/~cs2041/code/perl/code_examples.html>

i = 3
print("1. Enter line:")
last_line = sys.stdin.readline()
print("2. Enter line:")
for line in sys.stdin:
    if line == last_line:
        print("Snap!")
    last_line = line
    print("%s. Enter line:" % (i))
    i += 1
